tasks/solve-a-hidato-puzzle/src/main.rs

```diff
@@ -0,0 +1,106 @@
+use std::cmp::{max, min};
+
+#[derive(Clone, Debug)]
+struct Puzzle {
+    board: Vec<Vec<i32>>,
```

Comment:
I think it would be better to represent the board as a `Vec<Vec<Option<u32>>`.

Also, let's wrap the board implementation in a `Board` struct so that we can implement `Index<(usize, usize)>` on it.

Reply:
Good idea. I have tried to follow this, pls have a look.